### PR TITLE
CI: setup github actions to publish on PyPi

### PR DESCRIPTION
**Pull request by QKaiser**

### CODE REVIEW

**Comment by QKaiser** [comment]
cc @kissgyorgy since you reviewed #11

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  release:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Setup dependencies
+        uses: ./.github/actions/setup-dependencies
```
Comment (kissgyorgy) [inline .github/workflows/release.yml:16]: I would just inline the setup-dependencies action here, as it is not reusable in this repo.

File: .github/actions/setup-dependencies/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          ~/.cache/pypoetry/artifacts
+        key: poetry-cache-and-artifacts-${{ inputs.python-version }}
+
+    - name: Restore virtualenvs
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pypoetry/virtualenvs
+        key: venv-${{ hashFiles('poetry.lock') }}-${{ inputs.python-version }}
+
+    - name: Poetry install
```
Comment (kissgyorgy) [inline .github/actions/setup-dependencies/action.yml:39]: I think you don't need install before build.

**Review by kissgyorgy — CHANGES_REQUESTED** [review]

File: .github/actions/setup-dependencies/action.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Restore pip cache
```
Comment (kissgyorgy) [inline .github/actions/setup-dependencies/action.yml:15]: I'm not sure if these cache things are used for build?

**Review by kissgyorgy — APPROVED** [review]

File: .github/actions/setup-dependencies/action.yml (hunk 2 above)
Comment (kissgyorgy) [inline .github/actions/setup-dependencies/action.yml:39]: install is not needed for build

File: .github/actions/setup-dependencies/action.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ inputs.python-version }}
+
+    - name: Pip install
+      run: python -m pip install --upgrade pip poetry
+      shell: bash
+
+    - name: Restore poetry cache
```
Comment (kissgyorgy) [inline .github/actions/setup-dependencies/action.yml:25]: I don't think build is using any of the cache directories

File: .github/actions/setup-dependencies/action.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  python-version:
+    description: "Python version to setup"
+    required: false
+    default: "3.8"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Python
+      uses: actions/setup-python@v2
```
Comment (kissgyorgy) [inline .github/actions/setup-dependencies/action.yml:11]: I think this is the only action we need for publish, which can be inlined in `release.yml`

**Review by kissgyorgy — CHANGES_REQUESTED** [review]

File: .github/actions/setup-dependencies/action.yml (hunk 5 above)
Comment (QKaiser) [reply .github/actions/setup-dependencies/action.yml]: I inlined python and poetry installation in `release.yml`

File: .github/actions/setup-dependencies/action.yml (hunk 4 above)
Comment (QKaiser) [reply .github/actions/setup-dependencies/action.yml]: Indeed. Removed cache handling.

File: .github/actions/setup-dependencies/action.yml (hunk 2 above)
Comment (QKaiser) [reply .github/actions/setup-dependencies/action.yml]: Removed.

**Review by kissgyorgy — APPROVED** [review]

**Review by vlaci — APPROVED** [review]

**Pull request merged**